### PR TITLE
cpu/stm32l0: implement i2C non blocking mode

### PR DESCRIPTION
**Pull request by Marc-Aurele**
This PR implements the non blocking mode for I2C on stm32l0 family cpu. A timeout and a check of the error flags have been added for each "while" loop.

### CODE REVIEW

File: cpu/stm32l0/periph/i2c.c [hunk 1]
```diff
@@ -468,43 +500,68 @@ static void _start(I2C_TypeDef *dev, uint8_t address, uint8_t length, uint8_t rw
...
+    while (!(dev->ISR & I2C_ISR_TC) && tick--) {
+        if (dev->ISR & ERROR_FLAG || !tick)
+            break;
+    }
 
     /* send STOP condition */
     DEBUG("Generate stop condition\n");
     dev->CR2 |= I2C_CR2_STOP;
 }
-
```
Comment (aabadie) [inline cpu/stm32l0/periph/i2c.c]: Unrelated change

File: cpu/stm32l0/periph/i2c.c [hunk 2]
```diff
@@ -243,7 +252,8 @@ int i2c_read_bytes(i2c_t dev, uint8_t address, void *data, int length)
     _start(i2c, address, length, I2C_FLAG_READ);
 
     /* read the data bytes */
-    _read(i2c, data, length);
+    if (_read(i2c, data, length) < 0)
```
Comment (aabadie) [inline cpu/stm32l0/periph/i2c.c]: Please add opening and closing curly braces, even for one line blocks. Even if it's more verbose, this improves code readability.
I found this in other places of your changes.

Comment (Marc-Aurele) [reply cpu/stm32l0/periph/i2c.c]: done

File: cpu/stm32l0/periph/i2c.c (hunk 1 above)
Comment (Marc-Aurele) [reply cpu/stm32l0/periph/i2c.c]: done
